core/src/main/java/io/questdb/cairo/TableReaderMetadata.java

```diff
@@ -150,19 +153,22 @@ public void cloneTo(MemoryMA mem) {
     public void close() {
         Misc.free(metaMem);
         Misc.free(path);
+        transitionMeta = Misc.free(transitionMeta);
     }
 
-    public long createTransitionIndex() {
+    public long createTransitionIndex(long txnStructureVersion) {
         if (transitionMeta == null) {
             transitionMeta = Vm.getMRInstance();
         }
 
         transitionMeta.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
-        try (MemoryMR metaMem = transitionMeta) {
-            tmpValidationMap.clear();
-            TableUtils.validate(ff, metaMem, tmpValidationMap, ColumnType.VERSION);
-            return TableUtils.createTransitionIndex(metaMem, this.metaMem, this.columnCount, this.columnNameIndexMap);
+        tmpValidationMap.clear();
+        TableUtils.validate(transitionMeta, tmpValidationMap, ColumnType.VERSION);
+        if (txnStructureVersion != transitionMeta.getLong(TableUtils.META_OFFSET_STRUCTURE_VERSION)) {
```

Comment:
Can version check take place before the validation? The lighter the path - the better

Reply:
done in ed43dd5